sql: block interleaved tables with REGIONAL BY ROW

otan opened the pull request:
Release justification: Low-risk change to new functionality.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61437)
<!-- Reviewable:end -->

otan committed:
sql: block interleaved tables with REGIONAL BY ROW

Release justification: Low-risk change to new functionality.

Release note: None

ajstorm reviewed (approved):
Keep churning the butter.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61437#-:-MUz1L1pDqTY60FuNZ3s:b-n7allj)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
